src/model/wallet.ts

```diff
@@ -952,28 +1061,37 @@ export const createWallet = async (
 
 export const savePrivateKey = async (
   address: EthereumAddress,
-  privateKey: null | EthereumPrivateKey
+  privateKey: null | EthereumPrivateKey,
```

Comment:
note we are handling hardware wallet keys are being saved with publicAccess so we dont need auth to read them. 
I think this is a better UX flow then us requiring auth for this information and afaik it is not sensitive

Reply:
see my previous comment.